### DISCUSSION TITLE
Why do midi.controlChange and midi.programChange send midi notes instad of cc/pg?

### DISCUSSION THREAD

**Discussion by T-vK** [pschatzmann/arduino-midi#12] [Q&A]
I am using arduino-midi successfully to:
- receive midi notes (on and off)
- receive midi cc messages
- send midi notes (on and off)

But I'm failing at:
- sending midi cc messages
- sending program change messages

When I run:

``` C++
midi.controlChange(127, 0);
midi.programChange(0);
```

I receive:

```
Source  Event                  Ch  Data
 24:0   Reset
 24:0   Note off                0, note 0
 24:0   Note off                0, note 0
 24:0   Note off                0, note 127
 24:0   Reset
 24:0   Note off                0, note 0
 24:0   Note off                0, note 0
 24:0   Note off                0, note 0
 24:0   Note off                0, note 0
 24:0   Note off                0, note 0
```

But I would have expected this instead:
```
24:0   Control change          0, controller 127, value 0
24:0   Program change          0, program 0
```

If I just send basic notes (as in `midi.noteOn(1, 0);`) I am receiving the correct midi notes btw, so this issue only seems to be with control change and program change.

**Comment by pschatzmann** [comment]
Did you use for the sending and receiving both this library. 
If not try to test if this is working properly ?

**Reply by T-vK** [reply]
I have tried that and had the same issue.

I'm using ` midi->setDefaultSendingChannel(0);` btw.
And I can see that in noteOn and noteOff there is some special sauce around value `-1`:
```
void MidiCommon :: noteOn(uint8_t note, uint8_t velocity, int8_t channelPar) {
    uint8_t channel = channelPar != -1 ? channelPar : sendingChannel;
```

Is this missing from controlChange and programChange? Or is there a good reason for it not being there?

```
void MidiCommon :: controlChange(uint8_t msg, uint8_t value, int8_t channel){
    this->outMessage.status = 0b1011 << 4 | channel;
    this->outMessage.arg1 = msg;
    this->outMessage.arg2 = value;    
    writeData(&outMessage, 2);
}
```

`-1` is specified as the default in the header file:

```
virtual void controlChange(uint8_t msg, uint8_t value, int8_t channel=-1);
```

**Reply by pschatzmann** [reply]
Does it work properly if you add the channel to the method call ?

**Reply by T-vK** [reply]
It seems the issue was indeed caused by channel value `-1`. I have created a PR with a fix.
